src/spec-node/dockerCompose.ts

```diff
@@ -67,9 +73,12 @@ async function _openDockerComposeDevContainer(params: DockerResolverParameters,
 			// 	const featuresConfig = await generateFeaturesConfig(params.common, (await createFeaturesTempFolder(params.common)), config, async () => labels, getContainerFeaturesFolder);
 			// 	collapsedFeaturesConfig = collapseFeaturesConfig(featuresConfig);
 		}
-
-		const imageMetadata = getImageMetadataFromContainer(container, configWithRaw, undefined, idLabels, common.output).config;
-		const mergedConfig = mergeConfiguration(configWithRaw.config, imageMetadata);
+		const imageMetadata = applyConstraintsToMetadataEntries(
```

Comment:
Would it be easier to apply the constraints where the config / metadata is read? (E.g., in `internalGetImageMetadata()` and `readDevContainerConfigFile()`.) That would better ensure that we don't miss a case.

Reply:
I thought of that, had trouble initially piping in the `policyConstraintsP`, but i'm going to give it another look 👀

Reply:
Worked through the plumbing in https://github.com/devcontainers/cli/pull/771/commits/7feccee01d80ff690105bf8dccc901d454c0f5ec to run the policy earlier - is this closer to what you had in mind?

Reply:
Yes, great! Maybe `raw` in `getDevcontainerMetadata()` should keep the original data because that is written to the new image when you build one from a base image that already has metadata? Currently not a problem since `build` does not support a policy.

Reply:
7936feb51b9e5a0f4c8c1330db6551b97da34f66 makes changes to two sites preserving (not apply policy constraints) to `raw`.